In [8]:
import os

import numpy as np
import pandas as pd
import geopandas as gpd
import matplotlib.pyplot as plt

import xarray as xr
import rioxarray as rioxr

## 2. Fire perimeter data retrieval and selection

In [10]:
# Import California Fire Perimeter data
perimeter = gpd.read_file('data/California_Fire_Perimeters_(all).shp')
perimeter.head()

YEAR_ STATE AGENCY UNIT_ID  FIRE_NAME   INC_NUM  ALARM_DATE   CONT_DATE  \
0   2023    CA    CDF     SKU  WHITWORTH  00004808  2023-06-17  2023-06-17   
1   2023    CA    LRA     BTU     KAISER  00010225  2023-06-02  2023-06-02   
2   2023    CA    CDF     AEU    JACKSON  00017640  2023-07-01  2023-07-02   
3   2023    CA    CDF     AEU     CARBON  00018821  2023-07-11  2023-07-11   
4   2023    CA    CDF     AEU    LIBERTY  00018876  2023-07-11  2023-07-12   

   CAUSE  C_METHOD  OBJECTIVE  GIS_ACRES COMMENTS COMPLEX_NA  \
0      5         1          1    5.72913     None       None   
1      5         1          1   13.60240     None       None   
2      2         1          1   27.81450     None       None   
3      9         1          1   58.76020     None       None   
4     14         1          1   70.97900     None       None   

                                  IRWINID FIRE_NUM COMPLEX_ID  DECADES  \
0  {7985848C-0AC2-4BA4-8F0E-29F778652E61}     None       None     2020   
1  {43EBCC88-B3AC-48EB-8EF5-417FE0939CCF}     None       None     2020   
2  {B64E1355-BF1D-441A-95D0-BC1FBB93483B}     None       None     2020   
3  {CB41DB0A-E4B1-489D-A4EA-738F2CD6DB3B}     None       None     2020   
4  {F83F70A4-07A7-40B8-BD51-10CCC1C30D63}     None       None     2020   

                                            geometry  
0  POLYGON ((-13682443.000 5091132.739, -13682445...  
1  POLYGON ((-13576727.142 4841226.161, -13576726...  
2  POLYGON ((-13459243.000 4621236.000, -13458968...  
3  POLYGON ((-13468077.000 4642260.000, -13467975...  
4  POLYGON ((-13468418.000 4614853.000, -13468428...

a. The fire perimeter data was obtained from data.gov. The date of access was 

In [13]:
# Inspect the CRS
print('The CRS for the perimeter data is', perimeter.crs)
print('Is this CRS projected?', perimeter.crs.is_projected)

The CRS for the perimeter data is EPSG:3857
Is this CRS projected? True


In [18]:
# b. Select Thomas Fire boundary from perimeter data. Make sure to select the Thomas Fire that occured in 2017.
t_f_boundary = perimeter[(perimeter['FIRE_NAME'] == 'THOMAS') & (perimeter['YEAR_'] == 2017)]
t_f_boundary

YEAR_ STATE AGENCY UNIT_ID FIRE_NAME   INC_NUM  ALARM_DATE   CONT_DATE  \
2654   2017    CA    USF     VNC    THOMAS  00003583  2017-12-04  2018-01-12   

      CAUSE  C_METHOD  OBJECTIVE  GIS_ACRES                    COMMENTS  \
2654      9         7          1   281791.0  CONT_DATE based on Inciweb   

     COMPLEX_NA IRWINID FIRE_NUM COMPLEX_ID  DECADES  \
2654       None    None     None       None     2010   

                                               geometry  
2654  MULTIPOLYGON (((-13316089.016 4088553.040, -13...

## 3. True color image

In [2]:
fp = os.path.join('data','landsat8-2018-01-26-sb-simplified.nc')
landsat = rioxr.open_rasterio(fp)

In [3]:
landsat

<xarray.Dataset> Size: 25MB
Dimensions:      (band: 1, x: 870, y: 731)
Coordinates:
  * band         (band) int64 8B 1
  * x            (x) float64 7kB 1.213e+05 1.216e+05 ... 3.557e+05 3.559e+05
  * y            (y) float64 6kB 3.952e+06 3.952e+06 ... 3.756e+06 3.755e+06
    spatial_ref  int64 8B 0
Data variables:
    red          (band, y, x) float64 5MB ...
    green        (band, y, x) float64 5MB ...
    blue         (band, y, x) float64 5MB ...
    nir08        (band, y, x) float64 5MB ...
    swir22       (band, y, x) float64 5MB ...

The dimensions of the landsat data is (band:1, x:870, y:731). The variables included are the red, green, and blue band, as well as the near infrared and short wave infrared. 